sdk/servicebus/service-bus/samples/javascript/receiveMessagesStreaming.js

```diff
@@ -11,43 +12,71 @@
   Setup: Please run "sendMessages.ts" sample before running this to populate the queue/topic
 */
 
-const { delay, ServiceBusClient } = require("@azure/service-bus");
+const { delay, isMessagingError, ServiceBusClient } = require("@azure/service-bus");
 
 // Load the .env file if it exists
 require("dotenv").config();
 // Define connection string and related Service Bus entity names here
 const connectionString = process.env.SERVICE_BUS_CONNECTION_STRING || "<connection string>";
 const queueName = process.env.QUEUE_NAME || "<queue name>";
-async function main() {
+
+export async function main() {
   const sbClient = new ServiceBusClient(connectionString);
+
   // - If receiving from a subscription you can use the createReceiver(topic, subscription) overload
   // instead.
   // - See session.ts for how to receive using sessions.
   const receiver = sbClient.createReceiver(queueName);
 
-  const processMessage = async (brokeredMessage) => {
-    console.log(`Received message: ${brokeredMessage.body}`);
-    await brokeredMessage.complete();
-  };
-
-  const processError = async (args) => {
-    console.log(`Error from error source ${args.errorSource} occurred: `, args.error);
-  };
-
   try {
-    receiver.subscribe(
-      {
-        processMessage,
-        processError
+    const subscription = receiver.subscribe({
+      processMessage: async (brokeredMessage) => {
+        console.log(`Received message: ${brokeredMessage.body}`);
+
+        // autoComplete, which is enabled by default, will automatically call
+        // receiver.completeMessage() on your message so long as your
+        // processMessage handler does not throw an error.
+        //
+        // If your handler _does_ throw an error then the message will automatically
+        // be abandoned using receiver.abandonMessage()
+        //
+        // autoComplete can be disabled in the options for subscribe().
       },
-      {
-        autoComplete: false
+      processError: async (args) => {
+        console.log(`Error from source ${args.errorSource} occurred: `, args.error);
+
+        // the handler will not stop without explicit intervention from you.
```

Comment:
The "handler" would refer to what the user passes. We should refer to the client here.

```suggestion
        // the `subscribe() call will not stop trying to receive messages without explicit intervention from you.
```